### PR TITLE
rng-tools: 5 -> 6.6, jitterentropy: init at 2.1.2

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
New version :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by JohnAZoidberg** [comment]
Wait, don't merge yet. I think there's something wrong with the sources.

**Comment by JohnAZoidberg** [comment]
I had the source-hash wrong and it pulled an old source - probably the cache prevented it from failing.
Looks good now, builds with all configuration combinations and runs fine.

I also added the jitterentropy library to use it as a dependency.

**Comment by c0bw3b** [comment]
Thank you for tracking its new home and putting yourself as maintainer :+1:   (first things first)

File: pkgs/development/libraries/jitterentropy/default.nix
Comment (c0bw3b) [inline pkgs/development/libraries/jitterentropy/default.nix]: (nitpick) why the blank line here?

File: pkgs/development/libraries/jitterentropy/default.nix
Comment (c0bw3b) [inline pkgs/development/libraries/jitterentropy/default.nix]: Upstream repo rather suggest BSD3 or GPL**2** (not GPL3)

File: pkgs/tools/security/rng-tools/default.nix
Comment (c0bw3b) [inline pkgs/tools/security/rng-tools/default.nix]: (nitpick) why the blank line?

File: pkgs/tools/security/rng-tools/default.nix
Comment (c0bw3b) [inline pkgs/tools/security/rng-tools/default.nix]:
```suggestion
    ++ optional withGcrypt [ libgcrypt.dev ]
```

Otherwise the headers are not found during configurePhase

File: pkgs/tools/security/rng-tools/default.nix
Comment (c0bw3b) [inline pkgs/tools/security/rng-tools/default.nix]: (nitpick) Might be `gpl2Plus` since all the source files seems to include the following header:

```
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
```

But really not a blocker :smile_cat:

**Review by c0bw3b — CHANGES_REQUESTED** [review]
Mostly needs fixing the attribute name of jitterentropy

**Comment by c0bw3b** [comment]
And for broader comments:
* Consider adding `enableParallelBuilding = true` in both packages (these are not big builds but still)
* Is jitter entropy source considered experimental? Is it disabled by default upstream or was it your choice?

**Comment by JohnAZoidberg** [comment]
Thank you for the thorough review!

We we add `enableParallelBuilding = true` for all packages if it doesn't seem to cause problems?

No upstream enables everything by default if it sees that the dependency is installed. I disabled the nistbeacon because I don't want to make people's entropy pool unsuitable for cryptography without them knowing because that's probably what they would want to use this for.
Yeah, I thought about disabling jitterentropy by default, too, because I'm not entirely convinced it's numbers are of the same quality as the other hardware random sources. I I'll disable it and ask the maintainers of both projects if they think it's safe to use.

**Comment by c0bw3b** [comment]
> Do we add `enableParallelBuilding = true` for all packages if it doesn't seem to cause problems?

Yes. It's not the default because some packages break with this, but otherwise it's good practice to add it (after testing) and benefit from faster builds :)

Regarding configure flags I think your defaults are sane for now (RDRAND with libgcrypt enabled / NIST beacon and jitter entropy on-demand only). I was mostly curious. :cat2:

**Comment by JohnAZoidberg** [comment]
Alright :)
Why does ofborg-eval still fail? I can't seem to view the errors.
Is it okay that the no-aliases check fails or should I rename the libsysfs argument to sysfsutils?

**Review by c0bw3b — APPROVED** [review]

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build jitterentropy rng-tools

**Comment by JohnAZoidberg** [comment]
The attribute in `pkgs` is called `rng_tools` with an underscore... That's why ofborg can't evaluate your request.
We could change it and add an alias for the underscore, it looks like that is what we do for many other packages.

**Comment by c0bw3b** [comment]
Yes. If you're up for it it's indeed a good occasion the change the name to `rng-tools` in **pkgs/top-level/all-packages.nix**  and add an alias for `rng_tools` in **pkgs/top-level/aliases.nix**

And make this package compliant with the [naming guidelines](https://nixos.org/nixpkgs/manual/#sec-package-naming)

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build jitterentropy rng-tools

**Comment by c0bw3b** [comment]
@JohnAZoidberg I took the liberty to add a commit on top of yours to use the new pkg name in the `rngd` service module. The alias remains needed for out of tree users.

**Pull request merged**
